src/sync.cpp

```diff
@@ -96,25 +109,21 @@ static void push_lock(void* c, const CLockLocation& locklocation, bool fTry)
                 break;
 
             std::pair<void*, void*> p1 = std::make_pair(i.first, c);
-            if (lockorders.count(p1))
+            if (lockdata.lockorders.count(p1))
                 continue;
-            lockorders[p1] = (*lockstack);
+            lockdata.lockorders[p1] = (*lockstack);
 
             std::pair<void*, void*> p2 = std::make_pair(c, i.first);
-            if (lockorders.count(p2)) {
-                potential_deadlock_detected(p1, lockorders[p2], lockorders[p1]);
-                break;
```

Comment:
There's a difference between this and the upstream diff. Bitcoin adds the ``break`` statement in https://github.com/sipa/bitcoin/commit/660ff174f299a6dfb28f7725dfbb08f140a5474b#diff-ff1d9ab20c19185dbba33a95551fb6b2R91 , and Zcash adds it in https://github.com/bitcartel/zcash/commit/660ff174f299a6dfb28f7725dfbb08f140a5474b#diff-ff1d9ab20c19185dbba33a95551fb6b2R91 . But then in https://github.com/sipa/bitcoin/commit/9493803f4a60a234c2712bba000a6b9c93ffba94 , Bitcoin removes the ``break`` statement. We don't have the latter patch; do we need it (that is, does the upstream PR assume that it has been applied)?